sample-schemas/demo_schema.erl

```diff
@@ -0,0 +1,60 @@
+-module(demo_schema).
+
+-include_lib("typerefl/include/types.hrl").
+
+-type ip() :: ipv4() | ipv6().
+-type ipv4() :: {0..255, 0..255, 0..255, 0..255}.
+-type ipv6() :: {0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535}.
+
+-reflect_type([ip/0]).
+
+-export([keys/0]).
+
+keys() ->
+    [ {"foo.setting", fun setting/1}
+    , {"foo.endpoint", fun endpoint/1}
+    , {"foo.greet", fun greet/1}
+    ].
+
+setting(mapping) -> "app_foo.setting";
+setting(type) -> string();
+setting(converter) -> converter(string);
+setting(_) -> undefined.
+
+endpoint(mapping) -> "app_foo.endpoint";
+endpoint(type) -> ip();
+endpoint(converter) -> converter(ip);
+endpoint(_) -> undefined.
+
+greet(mapping) -> "app_foo.greet";
+greet(type) -> string();
+greet(validator) -> fun (Value) -> Value =:= "hello" end;
+greet(_) -> undefined.
+
+converter(ip) -> fun (Value) -> convert_to_ip(Value) end;
+converter(string) -> fun (Value) -> convert_to_string(Value) end.
+
+convert_to_ip(Bin) when is_binary(Bin) ->
+    convert_to_ip(binary_to_list(Bin));
+convert_to_ip(Str) when is_list(Str) ->
+    case inet:parse_address(Str) of
+        {ok, Addr} ->
+            Addr;
+        {error, einval} ->
+            Str
+    end;
+convert_to_ip(V) -> V.
+
+convert_to_string(Int) when is_integer(Int) ->
```

Comment:
`lists:flatten(io_lib:format("~p", [V]))` ?

Reply:
since we catch exception in `apply_converter`, now it is ok to be just `integer_to_list`